### ISSUE TITLE
[animations] Missing Support for SVG-related Animations

### ISSUE THREAD

**Issue by tomwanzek** [angular/angular#16427]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

The [Angular documentation](https://angular.io/docs/ts/latest/guide/animations.html) states:

> Since Angular's animation support builds on top of Web Animations, you can animate any property that the browser considers animatable. This includes positions, sizes, transforms, colors, borders, and many others. The W3C maintains a list of animatable properties on its CSS Transitions page.

These animatable properties include `svg` properties as per this [subsection 7.2](https://www.w3.org/TR/css3-transitions/#animatable-properties). A typical example would be the radius `r` of a `<svg:circle>` element.

Currently, the API for animations only supports a `style(...)`  property, which does not fully allow the animation of to be supported `svg` properties. 

Specifically, using the illustrative example component below, the animation will be performed in Chrome.
It will, however, fail in Firefox (and Edge). I.e. the end state will be immediately jumped to.

**Expected behavior**

Full support for all animatable `svg` properties in Angular on par with CSS style animation.
This could possibly involve adding a `attribute(...)` syntax as a complement to `style(...)` or expanding the supported use of the latter.

For added clarity, support for `svg` related animations should be a first class citizen in animation e.g. when addressing issues such as #9668 or any other means to creating dynamic animations.

Thanks in advance!!!

**Minimal reproduction of the problem with instructions**

The somewhat minimalistic component below allows reproduction.

```ts
import { Component } from '@angular/core';

import {
  trigger,
  state,
  style,
  animate,
  transition
} from '@angular/animations';

@Component({
  selector: 'app-root',
  template: `
    <svg width=100 height=100>
      <circle
        [@resize]
        cx=50
        cy=50
        r=20
      >
      </circle>
    </svg>
  `,
  styles: ['circle {stroke: slateblue; fill: slateblue;}'],
  animations: [
    trigger(
      'resize', [
        transition(
          'void => *', [
            style({ r: 0 }),
            animate(2000)
          ])
      ]

    )
  ]
})
export class AppComponent {

}
```

**What is the motivation / use case for changing the behavior?**

The primary use case are complex data visualizations using `svg` with animated changes in response to data changes.

Without adding in consistent support for `svg` related animations, it is necessary to utilize additional third party modules (e.g. `d3-selection` and `d3-transition`) to touch the DOM.

With full support for animations, D3 can be used to compute layout data, but as a segregated concern Angular can handle all DOM updates.

**Please tell us about your environment:**

Windows 10
VS Code 1.11.2
npm 3.10.10

* **Angular version:** Angular 4.1.0

* **Browser:** [| Chrome Version 57.0.2987.133 (64-bit) | Firefox 53.0 | MS Edge 40.15063.0.0 EdgeHTML 15.15063]

 
* **Language:** [all]

* **Node (for AoT issues):** n/a

**Comment by tomwanzek** [comment]
@marjan-georgiev @amcdnl wouldn't this be beneficial for [ngx-charts](https://github.com/swimlane/ngx-charts)?

**Comment by marjan-georgiev** [comment]
Absolutely. We still use d3-transition to animate some of the elements because we can not do it with @angular/animations. An example of that would be tweening the `path` `d` attribute. I have experimented with SMIL as a `hands off DOM` approach before, but with little success.

**Comment by tomwanzek** [comment]
One of the biggest impacts of this gap was noticing that, it reverberates all the way to the data structures and data flow between components. 

I.e. if I have to use `d3-selection`/`d3-transition` as part of life cycle hooks to support animations, the most idiomatic approaches imply using different `@Input()` data structures than if I could solely rely on Angular for DOM updates/change detection.

The most extreme example is to perform all DOM creation/manipulation _within_ a "D3" component using D3 just to get animation support.

**Comment by tomwanzek** [comment]
@matsko First of all, thanks for all the hard work on animations for Angular to date!!!

Given how immersed you are in the subject, could you please give a brief indication, if support for SVG attribute-related animations as considered above:
* will be added to Angular animations
* a ball-park time-frame for implementation, if on/added to the roadmap?

If the answer to the above is that it cannot/will not be done, a short rationale would be welcome too.

For me it is a fundamental question, as this is the **only** roadblock for a full separation of concerns between data visualization layout computation (D3) and actual DOM handling (Angular). Refactoring to use D3 to also touch the DOM within components is **material**: different templates, different data flow/data binding/change detection, different component code.

I would love to use dynamic data-driven animations directly in Angular, but if it will not be done (in the foreseeable future), a least I know that I have to commit to using D3 to handle everything inside my data visualization components.

Your guidance is most appreciated.

**Comment by tomwanzek** [comment]
@matsko Ping? This issue was opened in April and has not even seen any preliminary guidance/assessment from the core Angular Team.  Unfortunate...

cc @vicb

**Comment by tomwanzek** [comment]
@matsko @vicb Could someone kindly provide _at least_ an initial indication for the questions raised above? Would be more than appreciated. If you believe someone else should chime in, please do not hesitate to loop them in. Thx!!!

**Comment by tomwanzek** [comment]
@mhevery First of all, my thanks to the Angular Team for all the hard work... as always. In light of the fact that there always has been a strong commitment to community support: Is there anyone who can shed some perspective on the issue outlined above?

This has been a crossroads issue for a substantial amount of time now for me. There is material refactoring effort involved between using D3 to touch the DOM to support data-driven animations of SVG elements and, alternatively, using D3 only for layout calculations while Angular handles the DOM (including animations).

Clearly, I have a preference for separating the concerns and using Angular to handle the DOM, but if that is not possible for SVG-related animations, a clear guidance to that end would be alright.

If I have to abandon the use Angular animations, that would be unfortunate, but at least create planning certainty.

Thanks in advance for pointing me to an Angular contributor who can shed some light on the matter.

**Comment by nekkon** [comment]
Hey guys. Haven't tried animating a circle but if anyone is still interested in animating a svg you can animate g (group) elements. They can be used as a wrapper for circles or any other svg element you want. For example the code below works for me: 
````
<svg class="line" height="6" [attr.width]="lineWidth">
        <g [@lineAnimation]="data" (@lineAnimation.done)="onEnd()">
            <path [attr.fill]="color"
                d="M 0,2 C 0,0.66666663 0.66666663,0 2,0 h 37 c 1.333332,0 2,0.66666663 2,2 0,1.3333335 -0.666668,2 -2,2 H 2 C 0.66666663,4 0,3.3333335 0,2 Z"
                style="fill-opacity:1;fill-rule:nonzero;stroke:none"
                inkscape:connector-curvature="0" />
        </g>
    </svg>

I think this can be used as a workaround till properly fixed.

**Comment by tomwanzek** [comment]
Is there any update on this issue?

**Comment by mhevery** [comment]
This is blocked on Ivy.

**Comment by tomwanzek** [comment]
@mhevery Thanks for the swift clarification. Will it be moved from backlog to active development once Ivy is in?

**Comment by simondel** [comment]
Hi @mhevery Will this be fixed with Angular 9? Or will it still be blocked because libraries will be using VE instead of Ivy?

**Comment by kungufli** [comment]
Any update on this? If I had understood correctly, currently it's not possible to morph an SVG path using Angular animations, am I right?

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
